packages/@sanity/presence-store/src/PresenceStore.js

```diff
@@ -0,0 +1,152 @@
+/* global window */
+
+import Observable from '@sanity/observable/minimal'
+import UUID from '@sanity/uuid'
+import deepEqual from 'deep-equal'
+
+const RESEND_REPORT_INTERVAL = 15000
+const STATE_TIMEOUT_INTERVAL = RESEND_REPORT_INTERVAL * 2
+
+function hashKeyForState(identity, session) {
+  return `${identity}__${session}`
+}
+
+export default class PresenceStore {
+  constructor(connection, channel) {
+    this.connection = connection
+    this.unsubscribeListener = this.connection.listen().subscribe(this.handleMessage).unsubscribe
+    this.myState = {}
+    this.states = {}
```

Comment:
Perhaps use an ES6 `Map` for states and timestamps instead? It is more suitable for arbitrary keys, IIRC.